src/admin_api.ts

```diff
@@ -0,0 +1,374 @@
+import {
+    AccessKey,
+    AdminStats,
+    AdminTx,
+    Contact,
+    NewContact,
+    PaymailAddress,
+    SharedConfig,
+    Utxo,
+    XPub,
+    PageModel,
+    Webhook,
+    Metadata,
+    QueryPageParams,
+  } from './types';
+  
+  import {
+    AdminAccessKeyFilter,
+    AdminContactFilter,
+    AdminPaymailFilter,
+    AdminUtxoFilter,
+    TransactionFilter,
+    XpubFilter,
+  } from './filters';
+  
+  import { defaultLogger, Logger, LoggerConfig, makeLogger } from './logger';
+  import { HttpClient } from './httpclient';
+  import { buildQueryPath } from './query/query-builder';
+  
+  /**
+   * SPVWalletAdminAPI class for handling administrative operations
+   *
+   * @class SPVWalletAdminAPI
+   */
+  export class SPVWalletAdminAPI {
+    logger: Logger;
+    http: HttpClient;
+  
+    /**
+     * Creates a new instance of SPVWalletAdminAPI
+     *
+     * @param {string} serverUrl - The base URL of the SPV Wallet server
+     * @param {string} adminKey - Admin xPriv key
+     * @param {LoggerConfig} loggerConfig - Logger configuration (optional)
+     */
+    constructor(serverUrl: string, adminKey: string, loggerConfig: LoggerConfig = defaultLogger) {
```

Comment:
Maybe it should be similar to consturctor for user.
For example something like:
```suggestion
    constructor(serverUrl: string, options AdminClientOptions) {
```

where AdminClientOptions would be
```typescript
interface AdmincClientOptions extends XprivWithSigning {
   logger: LoggerConfig
}
```

_(maybe  `{ logger: LoggerConfig }` should also have it's type.)_

Reply:
Yes makes sense, keeps it consistent  with the userAPI, and allows extension of options later if required such as admin access key etc.